### PR TITLE
Add warning about using high precision mouse on non-windows platforms

### PR DESCRIPTION
**Pull request by peppy**
In response to #14021, just to provide awareness as I don't believe the underlying issue is going to be fixed in the near future.

### CODE REVIEW

**Comment by smoogipoo** [comment]
That issue is for linux though.

**Comment by peppy** [comment]
Missed that. We can add linux to the warning too. It's just as broken there, depending on your setup. Or can just leave this one as macOS.

**Comment by smoogipoo** [comment]
And you're sure this isn't just something we're/SDL is doing wrong? Like off by one px at the borders of the screen? Something that could affect Windows too?

**Comment by peppy** [comment]
It's per operating system issues. The implementation of high precision mode is different at an SDL level per OS and they all have different issues. This one is likely just incorrect warping (I already fixed a similar issue on windows upstream in SDL).

It's probably best we have a warning in place for non-windows OSes until we report and deal with each of the localised issues, to avoid users reporting them on an ongoing basis.

File: osu.Game/Overlays/Settings/SettingsItem.cs [hunk 1]
```diff
@@ -77,8 +82,8 @@ public string WarningText
                     });
                 }
 
-                warningText.Alpha = string.IsNullOrWhiteSpace(value) ? 0 : 1;
-                warningText.Text = value;
+                warningText.Alpha = hasValue ? 0 : 1;
+                warningText.Text = value.ToString(); // TODO: Add localisation support after https://github.com/ppy/osu-framework/pull/4603 is merged.
```
Comment (smoogipoo) [inline osu.Game/Overlays/Settings/SettingsItem.cs:86]: I don't understand this comment, is it linking to an incorrect PR?

File: osu.Game/Localisation/MouseSettingsStrings.cs [hunk 2]
```diff
@@ -54,6 +54,11 @@ public static class MouseSettingsStrings
         /// </summary>
         public static LocalisableString CursorSensitivity => new TranslatableString(getKey(@"cursor_sensitivity"), @"Cursor sensitivity");
 
+        /// <summary>
+        /// "This setting currently has issues on your platform. It is recommended to adjust sensitivity externally and keep this disabled for now."
```
Comment (smoogipoo) [inline osu.Game/Localisation/MouseSettingsStrings.cs:58]:
```suggestion
        /// "This setting has known issues on your platform. If you encounter problems, it is recommended to adjust sensitivity externally and keep this disabled for now."
```
https://github.com/ppy/osu-localisation-analyser/issues/37

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Comment by smoogipoo** [comment]
These settings probably shouldn't be enabled by default on these platforms, in that case.

**Comment by peppy** [comment]
That's [already the case for macOS](https://github.com/ppy/osu-framework/blob/652c3e054c5d93e37fe7d5a9442829c695f424f1/osu.Framework/Platform/MacOS/MacOSGameHost.cs#L67-L68). I'm open to also doing so on linux if it is seen as a majority case that's broken (but it doesn't seem that way based on the few reports of issues with it?).

**Comment by smoogipoo** [comment]
For me, I will probably never be enabling high precision as I run into the same as https://github.com/ppy/osu/issues/13684.

As I'm sure any Linux user who as much as uses XFCE/KDE with alt+drag to move their window will have the same issue.

**Review by smoogipoo — APPROVED** [review]

**Comment by peppy** [comment]
FWIW that same capture issue happens on windows if you attempt to do window dragging using hotkeys in a similar way.

**Comment by smoogipoo** [comment]
Sure, except in Windows it comes as a hack. In Linux it's built into most WMs so it comes as a pretty natural user interaction.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**